Src/WindowsServer/WindowsServer.Shared/Implementation/DataContracts/AzureInstanceComputeMetadata.cs

```diff
@@ -0,0 +1,177 @@
+﻿namespace Microsoft.ApplicationInsights.WindowsServer.Implementation.DataContracts
+{
+    using System;
+    using System.Globalization;
+    using System.Linq;
+    using System.Runtime.Serialization;
+    using System.Text.RegularExpressions;
+
+    /// <summary>
+    /// Class representing the returned structure from an Azure Instance Metadata request
+    /// for Compute information.
+    /// </summary>
+    [DataContract]
+    internal class AzureInstanceComputeMetadata
```

Comment:
https://github.com/Microsoft/ApplicationInsights-dotnet-server/tree/develop/Src/Common - @MS-TimothyMothra has made lot of helper classes already available here for sanitize etc. Re-use if applicable.

Reply:
Very interesting! I didn’t know about that group of utilities. I would like to push this out as a new issue though, to be handled in beta2…